pkg/controller/controller.go

```diff
@@ -38,6 +38,13 @@ import (
 	"k8s.io/client-go/util/retry"
 )
 
+// TCPPortMapper is capable of storing and retrieving a TCP port mapping for a given service.
+type TCPPortMapper interface {
+	Find(svc k8s.ServiceWithPort) (int32, bool)
+	Get(srcPort int32) *k8s.ServiceWithPort
+	Add(svc *k8s.ServiceWithPort) (int32, error)
+}
```

Comment:
That's a good observation! We have noticed mappings were never removed nor updated when receiving a Update events or a Delete event from an informer. This is definitely something that should be done. WDYT about fixing this issue in a different PR (Since this issue was already present in the code before)?

Reply:
Yup. Can you open an issue for that?

Reply:
An issue has been opened: #442